app/controllers/restrooms_controller.rb

```diff
@@ -16,7 +16,6 @@ def new
     if params[:edit_id]
       @restroom = find_restroom
       @restroom.edit_id = params[:edit_id]
-      @restroom.id = Restroom.last.id + 1
```

Comment:
I think without this `@restroom.save` will overwrite the existing entry, right? Since `find_restroom` returns the existing ID.

Reply:
Actually, all that the `new` method does is to help render the restroom submission page. The information being set here ends up as part of the form (`/app/views/restrooms/new`). The information in the form gets submitted to the `create` method, which creates a new restroom object from the parameters passed in.
```
@restroom = Restroom.new(permitted_params)
```
Restroom.new generates a new id automatically, so the old restroom isn't being overridden. Also, the old `id` isn't passed as part of `permitted_params` because there's no field for it in the form in the first place. You can test this locally to confirm.

Reply:
cool, lgtm to me then.